src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -781,16 +806,19 @@ private function syncBinaryData($nodeId, $binaryData)
         }
     }
 
-    private function syncReferences()
+    private function syncReferences($referencesToUpdate = null)
     {
-        if ($this->referencesToUpdate) {
+        $referencesToUpdate = $referencesToUpdate ? : $this->referencesToUpdate;
```

Comment:
yeah .. but we can just pass it in explicitly in all our calls to syncReferences() .. this would at least get rid of some of the state in this method.